shared/domains/chelonia/chelonia.js

```diff
@@ -812,6 +812,7 @@ export default (sbp('sbp/selectors/register', {
         'chelonia/private/in/syncContract', contractID, params
       ]).catch((err) => {
         console.error(`[chelonia] failed to sync ${contractID}:`, err)
+        sbp('okTurtles.events/emit', CONTRACT_IS_SYNCING, contractID, false)
```

Comment:
The proper way to fix this is by modifying `internals.js`, which is supposed to call `sbp('okTurtles.events/emit', CONTRACT_IS_SYNCING, contractID, false)`, and in fact already does in the `finally` block of that function. But the problem is that we don't get to the `finally` call, because the exception is thrown earlier at the call to

```js
const { HEAD: latestHEAD } = await sbp('chelonia/out/latestHEADInfo', contractID)
```

Which is before the `try` block even begins. So that is what needs to be fixed.

Reply:
As I've been working on this as well, I was able to get it work by modifying `internals.js` like so to move the `try` up:

```js
  'chelonia/private/in/syncContract': async function (contractID: string, params?: { force?: boolean, resync?: boolean }) {
    const state = sbp(this.config.stateSelector)
    try {
```

However, as you correctly pointed out on Slack, there is the additional problem that we should deal with, which is restoring the functionality of how it was working before when suddenly the identity contract has disappeared. We should forget about it, but right now we keep remembering:

```js
      sbp('gi.db/settings/load', SETTING_CURRENT_USER).then(async (identityContractID) => {
        // This loads CHELONIA_STATE when _not_ running as a service worker
        const cheloniaState = await sbp('gi.db/settings/load', 'CHELONIA_STATE')
        if (!cheloniaState || !identityContractID) return
        if (cheloniaState.loggedIn?.identityContractID !== identityContractID) return
        await sbp('chelonia/contract/sync', identityContractID, { force: true }) // this is throwing!
```

That `/sync` throws and whereas before we'd forget about the identity contract, now we are remembering it.

This logic is also questionable:

```js
        return sbp('gi.app/identity/login', { identityContractID }).catch((e) => {
          console.error(`[main] caught ${e?.name} while logging in: ${e?.message || e}`, e)
          console.warn(`It looks like the local user '${identityContractID}' does not exist anymore on the server 😱 If this is unexpected, contact us at https://gitter.im/okTurtles/group-income`)
        })
```

In that situation we would want to probably show a `gi.ui/prompt` to the user and to show the `e.message` that was caught.

So if you'd like to boyscout all of this in this PR as well, feel free!

Reply:
> That `/sync` throws and whereas before we'd forget about the identity contract, now we are remembering it.

It is probably OK to remember it between sessions, as it will be overwritten when we sign up with a new account (or login to another account that does work). Refining this can be done for another PR.

But this is probably still a good idea:

> In that situation we would want to probably show a `gi.ui/prompt` to the user and to show the e.message that was caught.

Reply:
added the error prompt for it.

![image](https://github.com/okTurtles/group-income/assets/17641213/128c3cd8-ed48-4a84-a829-252fcb15312f)

The thing I noticed is that this catch block never catches any log-in related errors.
```js
return sbp('gi.app/identity/login', { identityContractID }).catch((e) => {
          console.error(`[main] caught ${e?.name} while logging in: ${e?.message || e}`, e)
          console.warn(`It looks like the local user '${identityContractID}' does not exist anymore on the server 😱 If this is unexpected, contact us at https://gitter.im/okTurtles/group-income`)
```

So I placed the logic another catch block below it.